src/js/plugins/model.js

```diff
@@ -4,27 +4,46 @@ import { log } from 'utils/log';
 const pluginsRegistered = {};
 
 // Extract a plugin name from a string
-const getPluginName = function (url) {
+const getPluginName = function(url) {
     // Regex locates the characters after the last slash, until it encounters a dash.
     return url.replace(/^(.*\/)?([^-]*)-?.*\.(js)$/, '$2');
 };
 
-const PluginModel = function () {
-    this.addPlugin = function (url, fromLoader) {
-        const pluginName = getPluginName(url);
-        let plugin = pluginsRegistered[pluginName];
-        if (!plugin) {
-            plugin = new Plugin(url);
-            pluginsRegistered[pluginName] = plugin;
-        } else if (fromLoader && plugin.url !== url) {
-            log(`JW Plugin "${pluginName}" already loaded from "${plugin.url}". Ignoring "${url}."`);
+const PluginModel = function() {};
+const prototype = PluginModel.prototype;
+
+prototype.setupPlugin = function(url) {
+    const registeredPlugin = this.getPlugin(url);
+    if (registeredPlugin) {
+        if (registeredPlugin.url !== url) {
+            log(`JW Plugin "${getPluginName(url)}" already loaded from "${registeredPlugin.url}". Ignoring "${url}."`);
         }
-        return plugin;
-    };
+        return registeredPlugin.promise;
```

Comment:
See comment below - would we be able to `return Promise.resolve(registeredPlugin)`?

Reply:
No because the this could be a plugin whose request is still active. Resolving while the request is still in flight would cause `configurePlugin` to fail because the plugin class `plugin.js` is not yet available.

The test case for this is you setup a player, and while plugin requests are in flight, setup another player or cancel and resetup the same player.